### PR TITLE
Build process introduction, SASS precompilation

### PR DESCRIPTION
**Pull request by jonathanargentiero**
add new menu category Build management

### CODE REVIEW

File: build-process-introduction.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+## Table of contents
+
+- [Activating Gulp inside the Studio](#activating-gulp-inside-the-studio)
+- [Customize the build process](#customize-the-build-process)
+- [Additional modules](#additional-modules)
+
+## Activating Gulp inside the Studio
+
+[Gulp](http://gulpjs.com/) is a build tool based on [Node.js](https://nodejs.org/). It provide a Javascript interface to maintain complex build tasks intuitive and manageable.
```
Comment (hamzahik) [inline build-process-introduction.md]: @jonathanargentiero `It provide` -> `It provides`

File: build-process-introduction.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+---
+
+# Build process introduction
+
+## Table of contents
+
+- [Activating Gulp inside the Studio](#activating-gulp-inside-the-studio)
+- [Customize the build process](#customize-the-build-process)
+- [Additional modules](#additional-modules)
+
```
Comment (hamzahik) [inline build-process-introduction.md:11]: @jonathanargentiero for coherence it would be better to use or not `ing` form for all title :wink: `activate` , `customize` or `activating` , `customizing`

File: build-process-introduction.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  
+3. Restart the Studio  
+
+The console will inform you if Gulp is installed and correctly detected:
+
+<img src="img/console-gulp.png" />
+
+## Customize the build process
+
+Gulp logic can be summarized by the concept of *tasks*. A `gulpfile.js` can include multiple tasks. When Gulp is launched, it collects all the tasks defined by the user and run the speficied ones.
```
Comment (hamzahik) [inline build-process-introduction.md]: @jonathanargentiero `runs the specified ones`

**Comment by cedricss** [comment]
@jonathanargentiero 

> Titles

I prefer to homogenise the “style” of the titles, when possible (there are some practical exceptions like “Introduction”, “Prerequisites”…). For example, you could always start with a verb (without *ing) like this:
- Activate…
- Customize…
- Add more modules…

Same remark for the sass page.

> Gulp inside the Studio

The gulp setup check and installation should be in installation.html You should complet this page if information is missing then add a link to this page. This way the page would be shorter (shorter is better!).
The section would become “Introduction to Gulp” and only focus on gulpfile.js in mobile and web.

> “gulp serve is the command invoked by the Studio when Run Page is clicked"

This part may let think gulp is required to Run Page, whereas it’s just a bonus for Livereload, as stated in the installation.html and create-web-app.html pages.
I would prefer to present it this way: Gulp is optional, …, when installed you have several bonuses like Livereload, build tasks customisation, xxx, etc.

> These tasks are normally different for each project:

Add line break to fix the list formatting

At the end of the page, add a “going further” button link like at the end of the other pages. Example: “Let’s see an example: [Compile SASS files automatically »]”

> *Note: this functionality

Missing \* for format

> About the Sass page

Instead of explaining how to factorize the gulpfile “to make life easier”, I would prefer the project template to be ready for that (clean paths var ready to be completed, no more very long and hard to tweak gulp.watch task, etc). When reading the Sass precompilation example, the user should have the feeling “okay I just have to import this, add this path, add this line, and I’m done, Easy!”. The current version is a bit too long and hard to follow. And it’s not very clear I have to tweak on the web part, _then_ the mobile part. Maybe giving .patch examples (line added / edited) instead of .js one could be even more clear: "what should I patch?”.

File: sass-precompiling.md [hunk 4]
```diff
@@ -0,0 +1,292 @@
+---
+---
+
+# SASS precompiling
+
+## Table of contents
+
+- [Enable SASS precompiling](#add-sass-precompiling)
```
Comment (hamzahik) [inline sass-precompiling.md]: I think `precompilation` is more appropriate.

The same for the section `Add SASS precompiling` and its content

File: sass-precompiling.md [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+
+We will now see how to add SASS files precompiling to the build process. First we need to add to our `package.json` the following dependency:
+
+```javascript
+"devDependencies": {
+  "gulp-sass": "*"
+}
+```
+ 
+First set a variable that stores the source and the destination folders in the `gulpfile.js`:
```
Comment (hamzahik) [inline sass-precompiling.md]: we have already used `first` :wink: `First we need to add to our`package.json` `

File: sass-precompiling.md [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+gulp.task('sass', function(done) {
+  gulp.src(paths.sass.src) // define sources location
+    .pipe(sass()) // start sass process
+    .on('error', sass.logError) // halt on errors
+    .pipe(gulp.dest(paths.sass.dest)) // copy css into destination
+    .on('end', done); // quit
+});
+```
+
+Finally enable the `watch` task to survey the files changements under the `app/scss` folder with `sass` and `reload` tasks as second parameter:
```
Comment (hamzahik) [inline sass-precompiling.md]: `the files changements` -> `file changes`

**Comment by cedricss** [comment]
@jonathanargentiero 

Typos:

> Build page
- the speficied ones > specified
- the more popular packages > the most

> Sass page
- refractorization > refactorization
- takes in consideration > takes into
- Finally > Finally, 
- to survey the files > ("survey" seems incorrect) to look for any... (or equivalent)
- the files changements > the file changes

**Comment by cedricss** [comment]
Precompilation:
- Additional functionalities > should also be modified to start with a verb, if possible. It will be even better to be more precise than just "functionalities" (what kind? ...) 
- Add SASS precompilation part should mention it's about the web folder here
-  The same workflow for mobile and web > should also be modified. And the steps should be clearer. First we'll update the web part. Then the mobile part. The table of content should reflect that logic.
- _*Note: this functionality_:  (**x3**) missing a "*"
- _// 1 - change sass paths like this_ should be inside the code snippet

**Comment by jonathanargentiero** [comment]
just squashed commits

**Comment by cedricss** [comment]
Thanks @jonathanargentiero I merge this one, immediately followed by some fixes (d8e25b77a32403c1d0e2d4fae91d18fe7089fa5c)
